xarray/tests/test_coarsen.py

```diff
@@ -19,14 +19,14 @@
 from .test_dataset import ds
 
 
-def test_coarsen_absent_dims_error(ds) -> None:
+def test_coarsen_absent_dims_error(ds: Dataset) -> None:
     with pytest.raises(ValueError, match=r"not found in Dataset."):
         ds.coarsen(foo=2)
 
 
 @pytest.mark.parametrize("dask", [True, False])
 @pytest.mark.parametrize(("boundary", "side"), [("trim", "left"), ("pad", "right")])
-def test_coarsen_dataset(ds, dask, boundary, side) -> None:
+def test_coarsen_dataset(ds, dask, boundary, side):
```

Comment:
How come we had to remove these?

Reply:
coarsen reductions are not typed yet, see https://github.com/pydata/xarray/issues/6703.
Before, coarsed returned Any which does not complain.